docs/UserGuide.md

```diff
@@ -98,6 +99,7 @@ Keep an eye out for them!
 --------------------------------------------------------------------------------------------------------------------
 
 ## Features
+[Back to Table of Contents](#table-of-contents)
```

Comment:
I think it's better if we can put this at the end of every section. Actually I have added the link back to ToC in my other PR that is merged just now haha so I think we probably don't need this right now?